modules/nat-gateways/main.tf

```diff
@@ -9,21 +9,32 @@
  *
  */
 
+locals {
+  total_nat_count = var.enable_nat_creation ? var.nat_count : 0
+  total_new_nat   = var.enable_nat_creation ? (length(var.nat_eip) == 0 ? local.total_nat_count : 0) : 0
+  nat_ids         = var.enable_nat_creation ? (length(var.nat_eip) == 0 ? aws_eip.nat.*.id : values(data.aws_eip.nat)[*].id) : []
+}
```

Comment:
please add a comment above each local value to help future maintainers understand the intention/purpose of each of those.

Reply:
@ketzacoatl I have added comments to each of the variable and also renamed `nat_ids` to `eip_ids` for better clarity. Thanks for the review!